renpy/display/behavior.py

```diff
@@ -1192,7 +1200,12 @@ def __init__(self,
             if i.endswith("color"):
                 caretprops[i] = properties[i]
 
-        self.caret = renpy.display.image.Solid(xmaximum=1, style=style, **caretprops)
+        caret = renpy.display.image.Solid(xsize=1, style=style, **caretprops)
+        if caret_blink:
+            self.caret = renpy.store.Fixed(renpy.display.transform.Transform(caret, function=renpy.curry.curry(blink)(caret_blink=caret_blink)),
+                                           xsize=0,)
+        else:
+            self.caret = renpy.store.Fixed(caret, xsize=0)
```

Comment:
I think you can simply this section down to:
```suggestion
        if caret_blink:
            caret = renpy.display.transform.Transform(caret, function=renpy.curry.partial(blink, caret_blink=caret_blink))
    self.caret = renpy.store.Fixed(caret, xsize=0)
```